adam-cli/src/main/scala/org/bdgenomics/adam/cli/CompareADAM.scala

```diff
@@ -173,13 +173,19 @@ class CompareADAM(protected val args: CompareADAMArgs) extends ADAMSparkCommand[
 
   def run(sc: SparkContext, job: Job): Unit = {
 
-    if (args.listComparisons) {
+    // Work around for -list_comparisons (doesn't require @Arguments)
+    if (!args.listComparisons && (args.input1Path == null || args.input2Path == null)) {
+      val options = new CompareADAMArgs
+      val parser = new CmdLineParser(options)
+      println("\n2 Arguments (files) must be given as input.\n")
+      parser.printUsage(System.err)
+      sys.exit(1)
+    } else {
       println("\nAvailable comparisons:")
       DefaultComparisons.comparisons.foreach {
         generator =>
           println("\t%10s : %s".format(generator.name, generator.description))
       }
-
       return
```

Comment:
It terminate the run() method. Unless the caller has something else to do (which I don't think is the case) it will return also, and so on until the current thread's main run method returns. That will terminate the thread and if it is the last non-daemon thread, the equivalent of sys.exit(0) will be performed, i.e. the JVM process terminates with a zero status code.